test/model/message_test.dart

```diff
@@ -576,4 +603,197 @@ void main() {
     });
   });
+
+  group('handleSubmessageEvent', () {
+    late StreamMessage streamMessage;
+
+    Future<int> preparePollMessage({List<PollOption>? options}) async {
+      List<PollOption> effectiveOptions = options
+        ?? [PollOption(text: 'foo'), PollOption(text: 'bar')];
+      const messageId = 123;
+      streamMessage = eg.streamMessage(
+        id: messageId,
+        sender: eg.selfUser,
+      );
+      final submessages = [
+        eg.submessage(
+          senderId: eg.selfUser.userId,
+          content: jsonEncode(PollWidgetData(extraData: PollWidgetExtraData(
+            question: 'Favorite word',
+            options: effectiveOptions.map((e) => e.text).toList(),
+          )),
+        )),
+        for (int i = 0; i < effectiveOptions.length; i++)
+          ...[
+            for (final voter in effectiveOptions[i].voters)
+            eg.submessage(
+              senderId: voter,
+              content: jsonEncode(PollVoteEventSubmessage(
+                key: PollEventSubmessage.optionKey(senderId: null, idx: i),
+                op: PollVoteOp.add,
+              )),
+            )
+          ]
+      ];
+      await prepare();
+      await prepareMessageWithSubmessage(streamMessage, submessages);
+      return messageId;
+    }
+
+    test('message is unknown', () async {
+      await prepare();
+      await store.handleEvent(
+        eg.submessageEvent(1000, eg.otherUser.userId, content: '{}')
+      );
+      checkNotNotified();
+    });
+
+    test('message has no submessages', () async {
+      final streamMessage = eg.streamMessage(id: 123);
+      await prepare();
+      await prepareMessages([streamMessage]);
+      await store.handleEvent(
+        eg.submessageEvent(
+          streamMessage.id,
+          eg.otherUser.userId,
+          content: jsonEncode(PollQuestionEventSubmessage(question: 'New question'))));
+      checkNotNotified();
+      check(store.messages[123]).isNotNull().poll.isNull();
+    });
+
+    test('ignore submessage event with malformed content', () async {
+      final messageId = await preparePollMessage();
+      await checkLogs(() async {
+        await store.handleEvent(
+          eg.submessageEvent(
+            streamMessage.id,
+            eg.selfUser.userId,
+            content: jsonEncode({
+              'type': 'question',
+              // Invalid type for question
+              'question': 123,
+            })));
+      })..length.equals(2)
+        ..last.contains('Malformed submessage event data for poll');
+      checkNotNotified();
+      check(store.messages[messageId]).isNotNull().poll.isNotNull();
+    });
+
+    test('update question', () async {
+      final messageId = await preparePollMessage();
+      await store.handleEvent(
+        eg.submessageEvent(
+          streamMessage.id,
+          eg.selfUser.userId,
+          content: jsonEncode(PollQuestionEventSubmessage(question: 'New question'))));
```

Comment:
```suggestion
      final messageId = await preparePollMessage(question: 'Old question');
      await store.handleEvent(
        eg.submessageEvent(
          streamMessage.id,
          eg.selfUser.userId,
          content: jsonEncode(PollQuestionEventSubmessage(question: 'New question'))));
```

This way the test case is self-contained for the reader — the contrast between the original question value and the new question value is explicit within the test case's own source code.

(Similar to my comment above about `eg.pollWidgetDataFavoriteLetter`.)

Reply:
That's a great catch. I find it to be quite a good demonstration.